src/Core/Billing/Services/Implementations/SubscriberService.cs

```diff
@@ -330,4 +331,69 @@ await foreach (var paymentMethod in paymentMethods)
             }
         }
     }
+
+    public async Task<TaxInfo> GetTaxInformationAsync(ISubscriber subscriber)
+    {
+        ArgumentNullException.ThrowIfNull(subscriber);
+
+        if (string.IsNullOrWhiteSpace(subscriber.GatewayCustomerId))
+        {
+            logger.LogError("Cannot retrieve GatewayCustomerId for subscriber ({SubscriberID}) with no {FieldName}", subscriber.Id, nameof(subscriber.GatewaySubscriptionId));
+
+            return null;
+        }
+
+        var customer = await RetrieveCustomerAsync(subscriber);
```

Comment:
A method already exists for this: `GetCustomerOrThrow` within the `SubscriberService`.